python/pyspark/sql/pandas/serializers.py

```diff
@@ -155,7 +156,11 @@ def create_array(s, t):
             if t is not None and pa.types.is_timestamp(t):
                 s = _check_series_convert_timestamps_internal(s, self._timezone)
             try:
-                array = pa.Array.from_pandas(s, mask=mask, type=t, safe=self._safecheck)
+                if type(s.dtype) == CategoricalDtype:
+                    s = s.astype(s.dtypes.categories.dtype)
+                    array = pa.array(s)
```

Comment:
This doesn't need to be in the `try` block. Lets follow the above syntax where it checks if it is a timestamp type and do the following:
```python
elif type(s.dtype) == pd.CategoricalDtype:
  s = s.astype(s.dtypes.categories.dtype)
```
Also, it looks like this isn't even needed for pyarrow >= 0.16.1 as it will automatically cast to the right type. Could you add a note like "# This can be removed once minimum pyarrow version is >= 0.16.1"?

Reply:
Fixed